src/tfp_backend/Code_gen.ml

```diff
@@ -96,7 +96,11 @@ let pp_init ppf p =
   let pp_save_data ppf (name, st) =
     pf ppf "self.%s = %a" name (pp_cast "") (name, st)
   in
-  let ppbody ppf = (list ~sep:cut pp_save_data) ppf p.Program.input_vars in
+  let ppbody ppf =
+    if not (List.is_empty p.Program.input_vars) then
```

Comment:
Thanks for the patch!

I think this might be more clear as a `match`, what do you think?

```ocaml
  match p.Program.input_vars with
    [] -> pf ppf "pass"
    _ -> (list ~sep:cut pp_save_data) ppf p.Program.input_vars
```

Reply:
Sure, thanks!